src/transformers/quantizers/quantizer_torchao.py

```diff
@@ -80,11 +80,11 @@ def update_torch_dtype(self, torch_dtype):
                 logger.warning_once(
                     f"Setting torch_dtype to {torch_dtype} for int4_weight_only quantization, but only bfloat16 is supported right now. Please set the torch_dtype to bfloat16."
                 )
-            if torch_dtype is None:
-                logger.warning_once(
-                    "Setting torch_dtype to torch.bfloat16 for int4_weight_only quantization since only bfloat16 is supported right now. Please set torch_dtype=torch.bfloat16 to remove this warning."
-                )
-                torch_dtype = torch.bfloat16
+        if torch_dtype is None:
+            logger.warning_once(
+                "Setting torch_dtype to torch.bfloat16 for int4_weight_only quantization since only bfloat16 is supported right now. Please set torch_dtype=torch.bfloat16 to remove this warning."
+            )
+            torch_dtype = torch.bfloat16
```

Comment:
Please leave the indent. This is only the case for int4_weight_only no ?

Reply:
I found the dtype is None at some point actually, so it feels like we'd need to set torch_dtype here?

Reply:
The dtype will be set to torch.float32 later on if the user don't specify `torch_dtype`. Let's leave the following code only for `int4_weight_only`

```py
        if torch_dtype is None:
            logger.warning_once(
                "Setting torch_dtype to torch.bfloat16 for int4_weight_only quantization since only bfloat16 is supported right now. Please set torch_dtype=torch.bfloat16 to remove this warning."
            )
            torch_dtype = torch.bfloat16
```